site/app/templates/forum/ForumBar.twig

```diff
@@ -59,42 +64,44 @@
 			{% if user_group <= button.required_rank %}
 				{% set link = (button.link[0]) ? 'href=' ~ button.link[1] %}
 				{% set onclick = (button.onclick[0]) ? 'onclick=' ~ button.onclick[1] %}
-				<a class="btn btn-primary {{ button.optional_class }}" style="{{ button.style }}" {{ link }} title="{{ button.title }}" {{ onclick }} >
+				<a tabindex="{{ buttonCounterRight }}" class="btn btn-primary {{ button.optional_class }}" style="{{ button.style }}" {{ link }} title="{{ button.title }}" {{ onclick }} >
 					{{ button.display_text|raw }}
 				</a>
+				{% set buttonCounterRight = buttonCounterRight + 1 %}
 			{% endif %}
 		{% endfor %}
 
-	{% if show_more %}
-	    <div class="dropdown" style="float:right;top:3px;margin-right:5px;">
-	        <a title="More options" id="dropdownLabel" role="button" data-toggle="dropdown" class="btn btn-default dropdown-toggle" data-target="#" href="#">
-	            More <span class="caret"></span>
-	        </a>
-			<ul class="dropdown-menu multi-level" role="menu" aria-labelledby="dropdownMenu">
-			{% for more in more_data|slice(1) %}
-				{% if user_group <= more.required_rank %}
-					{% set onclick = (more.onclick[0]) ? 'onclick=' ~ more.onclick[1] %}
-					{% set optional_class = (more.optional_class[0]) ? 'class=' ~ more.optional_class[1] %}
-	          		<li class='dropdown-item'><a id="{{ more.id }}" title="{{ more.title }}" {{ optional_class }} href="{{ more.link }}" {{ onclick }}>{{ more.display_text }}</a></li>
-	          	{% endif %}
-	        {% endfor %}
-	          <li class="dropdown-divider"></li>
-	              <li title="Sort posts by reply hierarchy" id="tree" class='dropdown-item'><a tabindex="-1" onclick="changeDisplayOptions('tree', {{ current_thread }})" href="#">Hierarchical</a></li>
-	              <li title="Sort posts by ascending chronological order" id="time" class='dropdown-item'><a href="#" onclick="changeDisplayOptions('time', {{ current_thread }})">Chronological  <i class="fas fa-angle-up"></i> </a></li>
-				  <li title="Sort posts by descending chronological order" id="reverse-time" class='dropdown-item'><a href="#" onclick="changeDisplayOptions('reverse-time', {{ current_thread }})">Chronological  <i class="fas fa-angle-down"></i> </a></li>
-	              {% if user_group <= core.getUser().accessGrading() %}
-	              	<li title="Sort posts by author's last name" id="alpha" class='dropdown-item'><a href="#" onclick="changeDisplayOptions('alpha', {{ current_thread }})">Alphabetical</a></li>
-	              {% endif %}
-	        </ul>
-	    </div>
-	{% endif %}
+		{% if show_more %}
+			<div class="dropdown" style="float:right;top:3px;margin-right:5px;">
+				<a tabindex="{{ buttonCounter }}" title="More option" id="dropdownLabel" role="button" data-toggle="dropdown" class="btn btn-default dropdown-toggle" data-target="#" href="#">
+					More <span class="caret"></span>
+				</a>
+				{% set buttonCounter = buttonCounter + 1 + buttonCounterRight %}
```

Comment:
It works in my browser, but I am not sure what this line means. `buttonCounter` doesn't seem to be used in the code below. Why does it need to be set here, and what does `buttonCounter + 1 + buttonCounterRight` mean?

Reply:
Ah, that's a quick workaround to the fact that on the right, the buttons are in `float: right` CSS, and that makes the button on the left actually being on the right...you know how float-right works! So for all the right floated buttons, count them up separately, and then add that back to the next item, so that the tab indexing goes correctly in the view!